back-end/src/lms/views/students.py

```diff
@@ -176,8 +189,114 @@ def application(self, request: Request, pk=None) -> Response:
             data=ApplicationProcessSerializer(instance=updated).data,
         )
 
-
-@extend_schema(tags=["students"])
+    def generate_excel(self, students):
+        """Generate an Excel file with information about the students."""
+
+        workbook = xlsxwriter.Workbook('/tmp/export.xlsx')
```

Comment:
Why not use https://docs.python.org/3/library/tempfile.html?

Reply:
This thing accepts only strings, not file objects https://xlsxwriter.readthedocs.io/workbook.html#constructor